scripts/getPVAliases.py

```diff
@@ -0,0 +1,342 @@
+    print(f'{Fore.LIGHTGREEN_EX}Found the following:{Style.RESET_ALL}')
+    print(build_table(temp, col_list))
+
+
+###############################################################################
+# %% Argparser
+###############################################################################
+# parser obj configuration
+parser = argparse.ArgumentParser(
+    prog='gatherPVAliases',
+    description="gathers all record <-> alias associations from a child's "
+                "ioc.cfg, st.cmd, and parent ioc.cfg.",
+                epilog='')
+# main command arguments
+parser.add_argument('patt', type=str)
+parser.add_argument('hutch', type=str)
+parser.add_argument('-d', '--dry_run', action='store_true',
+                    default=False,
+                    help="Forces a dry run for the script. "
+                    "No files are saved.")
+
+###############################################################################
+# %% Main
+###############################################################################
+
+
+def main():
+    """
+    Main function entry point
+    """
+    # parse args
+    args = parser.parse_args()
+    # search ioc_cfg and build the dataset
+    data = find_ioc(args.hutch, args.patt)
+    if data is None:
+        print(f'{Fore.RED}No results found for {Style.RESET_ALL}{args.patt}'
+              + f'{Fore.RED} in{Style.RESET_ALL}'
+              + f'{args.hutch}')
+        sys.exit()
+
+    # find the parent directories
+    for _d in data:
+        _d['parent_ioc'] = find_parent_ioc(_d['id'], _d['dir'])
+
+    # Hard code the column order for the find_ioc output
+    column_list = ['id', 'dir',
+                   Fore.LIGHTYELLOW_EX + 'parent_ioc' + Style.RESET_ALL,
+                   'host', 'port',
+                   Fore.LIGHTBLUE_EX + 'alias' + Style.RESET_ALL,
+                   Fore.RED + 'disable' + Style.RESET_ALL]
+    if args.hutch == 'all':
+        column_list = ['hutch'] + column_list
+    show_temp_table(data, column_list)
+
+    ans = simple_prompt('Proceed? (Y/n): ', default='Y')
+    # Abort if user gets cold feet
+    if ans is False:
+        sys.exit()
+    print(f'{Fore.RED}Skipping disabled child IOCs{Style.RESET_ALL}')
+
+    # initialize the final output to write to file
+    final_output = []
+
+    # iterate through all the child ioc dictionaries
+    for _ioc in data:
+        if _ioc.get('disable') is not True:
+            # first acquire the base alias dictionary
+            alias_dicts = acquire_aliases(_ioc['dir'], _ioc['id'])
+            # show the record aliases to the user
+            print(Fore.LIGHTGREEN_EX
+                  + 'The following substitutions were found in the st.cmd:'
+                  + Style.RESET_ALL)
+            print(build_table(alias_dicts, ['record', 'alias'], align='l'))
+            # optional skip for all resulting PV aliases
+            save_all = (simple_prompt(
+                'Do you want to save all resulting PV aliases? '
```

Comment:
I should update this docstring to "Do you want to add all resulting PV <--> alias associations to the final output?"

The script now saves a single file with all the results appended together (less annoying to parse when you're playing with the result later). Since you're iterating through a base PV and its resulting PVs (each with aliases), instead of pestering the user to approve every single instance (e.g. each axis in an MCS2 ioc), the user can approve if the first association looks good.

Reply:
I think if the script saves a file we should mention this in the docstring.

Reply:
More informative?
![image](https://github.com/pcdshub/engineering_tools/assets/149725219/d6d6578a-2ea8-4615-9c45-08a19e9de668)

Reply:
It's definitely better but I think my overall confusion was a lack of understanding that this script was supposed to save anything at all.

Reply:
With some massaging on `argparse.RawTextHelpFormatter`:
![image](https://github.com/pcdshub/engineering_tools/assets/149725219/d70c8f44-3ae3-49a8-9a5a-2688a7d3a71a)

Reply:
Looks good to me! Very helpful output!